File: seeker/snippet/rectangle.py
#date: 2021-11-05T16:49:17Z
#url: https://api.github.com/gists/19895add83c5cd762d18e6721a2473bb
#owner: https://api.github.com/users/treyhunner

class Rectangle:

    def __init__(self, width, height):
        self.width, self.height = width, height

    def __repr__(self):
        return f"Rectangle({self.width}, {self.height})"
 
    @property
    def area(self):
        return self.width * self.height

    @area.setter
    def area(self, area):
        self.width *= area/self.area

File: seeker/snippet/Rectangle.java
//date: 2024-05-21T16:51:08Z
//url: https://api.github.com/gists/06a2af3079c7654e758daf457ab8379a
//owner: https://api.github.com/users/maulanafanny

public class Rectangle implements Quadrilateral {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public double calculateArea() {
        return width * height;
    }

    @Override
    public double calculatePerimeter() {
        return 2 * (width + height);
    }
}